run.go

```diff
@@ -43,6 +43,9 @@ func Run(in context.Context, s Spec) error {
 		opts []StreamOption
 	)
 	for _, opt := range s.opts {
+		if opt == nil {
+			panic("nil opt passed in")
```

Comment:
We were weighing up whether to return an error here too. 
In the Luno context, this wouldn't cause the deployment to roll back (I think). But I guess we could do an error check when we call it, and panic `if err != nil`

Reply:
Thinking about it, I'm quite reluctant to panic in this context. I'm thinking we should return a sentinel error instead